WeatherXM/wxm-android
# FE-502 Add more Detekt rules and fix the issues arised

## **Why?**

More Detekt rules to have a better and more consistent code styling.

### **How?**
Added the following:

**Style Rules:**
1. [AlsoCouldBeApply](https://detekt.dev/docs/rules/style/#alsocouldbeapply)
2. [CanBeNonNullable](https://detekt.dev/docs/rules/style/#canbenonnullable)
3. [CascadingCallWrapping](https://detekt.dev/docs/rules/style/#cascadingcallwrapping)
4. [CollapsibleIfStatements](https://detekt.dev/docs/rules/style/#collapsibleifstatements)
5. [NullableBooleanCheck](https://detekt.dev/docs/rules/style/#nullablebooleancheck)
6. [OptionalUnit](https://detekt.dev/docs/rules/style/#optionalunit)
7. [RedundantExplicitType](https://detekt.dev/docs/rules/style/#redundantexplicittype)
8. [SpacingBetweenPackageAndImports](https://detekt.dev/docs/rules/style/#spacingbetweenpackageandimports)
9. [UnnecessaryApply](https://detekt.dev/docs/rules/style/#unnecessaryapply)
10. [UnnecessaryBracesAroundTrailingLambda](https://detekt.dev/docs/rules/style/#unnecessarybracesaroundtrailinglambda)
11. [UnnecessaryLet](https://detekt.dev/docs/rules/style/#unnecessarylet)
12. [UnusedImports](https://detekt.dev/docs/rules/style/#unusedimports)
13. [UseDataClass](https://detekt.dev/docs/rules/style/#usedataclass)
14. [UseIfEmptyOrIfBlank](https://detekt.dev/docs/rules/style/#useifemptyorifblank)
15. [UseIfInsteadOfWhen](https://detekt.dev/docs/rules/style/#useifinsteadofwhen)
16. [UseLet](https://detekt.dev/docs/rules/style/#uselet)

**Performance:**
1. [CouldBeSequence](https://detekt.dev/docs/rules/performance#couldbesequence)

**Naming:**
1. [VariableMaxLength](https://detekt.dev/docs/rules/naming#variablemaxlength)
2. [VariableMinLength](https://detekt.dev/docs/rules/naming#variableminlength)

### **Testing**

1. Build the project
18. run `gradle detekt` to ensure no issues arise
